kerl

```diff
@@ -754,7 +668,133 @@ _flags() {
+_KPP_debian_probe="_dpkg \$1"
+
+_KPP_fedora="openssl-devel make automake autoconf ncurses-devel gcc g++"
+_KPP_fedora_probe="_rpm \$1"
+
+_KPP_linuxmint="libssl-dev make automake autoconf libncurses5-dev gcc g++"
+_KPP_linuxmint_probe="_dpkg \$1"
+
+_KPP_opensuse="openssl-devel make automake autoconf ncurses-devel gcc g++"
+_KPP_opensuse_probe="_rpm \$1"
+
+_KPP_rhel="openssl-devel make automake autoconf ncurses-devel gcc g++"
+_KPP_rhel_probe="_rpm \$1"
+
+_KPP_ubuntu="libssl-dev make automake autoconf libncurses5-dev gcc g++"
+_KPP_ubuntu_probe="_dpkg \$1"
+
+parse_os_release() {
+    # $1: path to os-release
+    # returns:
+    # - 2 if release ID is not found in os-release file
+    os_release_id0=$(\grep "^ID=" "$1")
+    os_release_id=$(echo "${os_release_id0}" | \sed 's|.*=||')
+    os_release_pretty_name0=$(\grep "^PRETTY_NAME=" "$1")
+    os_release_pretty_name=$(echo "${os_release_pretty_name0}" | \sed 's|.*=||')
+    echo "[packages] Found ${os_release_pretty_name} with ID ${os_release_id}" >>"$LOGFILE"
+    if [ -z "${os_release_id}" ]; then
+        echo "[packages] Release ID not found in $1" >>"$LOGFILE"
+        echo "2"
+    else
+        echo "[packages] Release ID found in $1: ${os_release_id}" >>"$LOGFILE"
+        echo "${os_release_id}"
+    fi
+}
+
+get_id_from_os_release_files() {
+    # returns:
+    # 1 - if no release files exist
+    files="/etc/os-release /usr/lib/os-release"
+    for file in ${files}; do
+        if [ -f "${file}" ]; then
+            parsed=$(parse_os_release "${file}")
+            if [ "${parsed}" != "2" ]; then
+                break
+            fi
+        fi
+    done
+
+    if [ -z "${parsed}" ]; then
+        echo "[packages] Found no file in: ${files}. Bailing out..." >>"$LOGFILE"
+        echo "1"
+    fi
+
+    echo "${parsed}"
+}
+
+probe_pkgs() {
+    os_release_id=$(get_id_from_os_release_files)
+
+    if [ "${os_release_id}" = "1" ]; then
+        msg="[packages] Unable to determine Linux flavor (no release files); not checking build packages."
+        echo "${msg}" >>"$LOGFILE"
+        l=w stderr "${msg}"
+        return
+    elif [ "${os_release_id}" = "2" ]; then
+        msg="[packages] Unable to determine Linux flavor (no ID); not checking build packages."
+        echo "${msg}" >>"$LOGFILE"
+        l=w stderr "${msg}"
+        return
+    fi
+
+    kpp=$(eval echo \$_KPP_"${os_release_id}"_pkgs)
+    if [ -n "${kpp}" ]; then
+        echo "[packages] Found package declarations for ${os_release_id}. Linux flavor is known." >>"$LOGFILE"
+        echo "[packages] Acting on known Linux distro ${os_release_id}" >>"$LOGFILE"
+        for pkg in ${kpp}; do
+            cmd=$(eval echo "\$_KPP_${os_release_id}_probe")
+            probe_without_1=$(echo "${cmd}" | sed "s|\$1||")
+            if [ "${probe_without_1}" != "${cmd}" ]; then
+                cmd=$(echo "${cmd}" | sed "s|\$1|${pkg}|")
+            fi
```

Comment:
This is mentioned above, and gives us more flexibility. Since we're evaluating strings, I figured we need a way to call internal functions. This'll most likely be the regular use case.